### PR TITLE
Cleaned up the argumentlist and checked for exit code

### PR DESCRIPTION
**Pull request by jwmoss**
Happy Hacktoberfest :)

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Thanks! I will cut #118 first to stabilize the images, but then I will review it and try to get integrated.

P.S: Hacktoberfest starts on Oct 1, but we appreciate all pull requests anyway

**Comment by jwmoss** [comment]
After I submitted the PR I remembered that Hacktoberfest hadn’t begun, but oh well!

File: jenkins-agent.ps1 [hunk 1]
```diff
@@ -108,11 +108,33 @@ if(![System.String]::IsNullOrWhiteSpace($Cmd)) {
...
+	)
 
 	#TODO: Handle the case when the command-line and Environment variable contain different values.
-	#It is fine it blows up for now since it should lead to an error anyway.
-	Start-Process -FilePath $JAVA_BIN -Wait -NoNewWindow -ArgumentList $("$jnlpProtocolOpts -cp C:/ProgramData/Jenkins/agent.jar hudson.remoting.jnlp.Main -headless$Tunnel$Url$WorkDir$Secret $Name")
+
+	Write-Verbose "Starting to execute Jenkins Agent"
+	$process = Start-Process -FilePath $JAVA_BIN -Wait -NoNewWindow -ArgumentList $ArgumentList
+	$exitCode = $process.ExitCode
+	if ($exitCode -eq 0)
```
Comment (slide) [inline jenkins-agent.ps1]: I don't think the process will ever return.

Comment (slide) [reply jenkins-agent.ps1]: Also, without the -PassThru arguement to `Start-Process`, a Process object is not returned:

See https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/start-process?view=powershell-6#outputs

File: jenkins-agent.ps1 [hunk 2]
```diff
@@ -108,11 +108,33 @@ if(![System.String]::IsNullOrWhiteSpace($Cmd)) {
...
+		$Secret,
+		$Name
+	)
 
 	#TODO: Handle the case when the command-line and Environment variable contain different values.
-	#It is fine it blows up for now since it should lead to an error anyway.
-	Start-Process -FilePath $JAVA_BIN -Wait -NoNewWindow -ArgumentList $("$jnlpProtocolOpts -cp C:/ProgramData/Jenkins/agent.jar hudson.remoting.jnlp.Main -headless$Tunnel$Url$WorkDir$Secret $Name")
+
+	Write-Verbose "Starting to execute Jenkins Agent"
+	$process = Start-Process -FilePath $JAVA_BIN -Wait -NoNewWindow -ArgumentList $ArgumentList
```
Comment (jglick) [inline jenkins-agent.ps1]: This seems to produce an error as noted in https://github.com/jenkinsci/kubernetes-plugin/pull/626#issuecomment-545222571.

Comment (jwmoss) [reply jenkins-agent.ps1]: @jglick Is you type `$ArgumentList` when running this when it fails, what's in it? I'm wondering if it's extra whitespace or something else.

Comment (jglick) [reply jenkins-agent.ps1]: I did not type anything related. Just (apparently) used the image from this PR by accident, and got a cryptic error. Switching to the regular image (no other changes) fixed it. That is all I know.

**Comment by slide** [comment]
I think this is superseded by #154

**Pull request closed without merging**